independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenRepoInitializer.java

```diff
@@ -218,6 +273,14 @@ public static DefaultRepositorySystemSession newSession(RepositorySystem system,
         return remotes;
     }
 
+    private static Profile getProfile(String name, Settings settings) throws AppModelResolverException {
+        final Profile profile = settings.getProfilesAsMap().get(name);
+        if(profile == null) {
+            throw new AppModelResolverException("Failed to locate profile " + name + " in Maven settings");
```

Comment:
I think Maven itself logs a warning when you have an unrecognized profile. Isn't throwing an exception a bit too harsh?